modules/apps/frontend-taglib/frontend-taglib-clay/src/main/resources/META-INF/resources/management_toolbar/ResultsBar.js

```diff
@@ -42,32 +75,31 @@ const ResultsBar = ({
 				>
 					<span
 						aria-label={sub(
-							itemsTotal === 1
-								? Liferay.Language.get('x-result-for-x')
-								: Liferay.Language.get('x-results-for-x'),
-							[
+							getResultText(
 								itemsTotal,
-								filterLabelItems
-									?.map((item) => item.label)
-									.join(', '),
-							]
+								filterLabelItems?.length || 0
+							),
+							itemsTotal,
+							`"${searchValue}"`
 						)}
 						className="component-text text-truncate-inline"
 						ref={resultsBarRef}
 						tabIndex={-1}
 					>
-						<span className="text-truncate">
-							{sub(
-								itemsTotal === 1
-									? Liferay.Language.get('x-result-for')
-									: Liferay.Language.get('x-results-for'),
-								itemsTotal
-							)}
-
-							{searchValue && (
-								<strong>{` "${searchValue}"`}</strong>
-							)}
-						</span>
+						<span
+							className="text-truncate"
+							dangerouslySetInnerHTML={{
+								__html: sub(
+									getResultText(
+										searchValue,
+										itemsTotal,
+										filterLabelItems?.length || 0
+									),
+									itemsTotal,
+									`<strong>"${searchValue}"</strong>`
+								),
+							}}
```

Comment:
This use of `dangerouslySetInnerHTML` is a bit of a worry for me. Is there any way we can avoid it? And by passing in user generated content, I fear we might have a possible XSS issue here.

Reply:
I looked into this a bit, I think it is because the `sub` function doesn't actually support jsx... However, there are a few examples in DXP where it does work with JSX. [See this sub util for example](https://github.com/liferay/liferay-portal/blob/master/modules/apps/item-selector/item-selector-taglib/src/main/resources/META-INF/resources/utils/utils.es.js#L11)

I'm not positive if we can simply update our sub util from frontend-js-web though. If you can take a look, that'd be great. Otherwise, can you just copy/paste the other one I linked and use that for now? And if you do that, then also create an LPS that we can use to unify our use of `sub` across DXP to work for all legacy cases and also JSX use case.

Thanks!

Reply:
Oh sure, I'll have a look to update it. Great idea :smile:

Reply:
Updated! This looks much better, thanks for the idea :smile: 

![image](https://github.com/liferay-frontend/liferay-portal/assets/800645/d53c4fd5-45e7-45ad-88bc-54d5bbd06ad9)